app/src/main/java/com/ciandt/dragonfly/example/features/download/DownloadIntentService.kt

```diff
@@ -0,0 +1,86 @@
+package com.ciandt.dragonfly.example.features.download
+
+import android.app.IntentService
+import android.content.Context
+import android.content.Intent
+import com.ciandt.dragonfly.example.BuildConfig
+import com.ciandt.dragonfly.example.helpers.NotificationHelper
+import com.ciandt.dragonfly.example.infrastructure.extensions.getDownloadManager
+import java.io.File
+import java.io.FileOutputStream
+
+class DownloadIntentService : IntentService("DownloadIntentService") {
+
+    override fun onHandleIntent(intent: Intent?) {
+
+        if (intent == null) {
+            return
+        }
+
+        val downloadedFile = intent.getParcelableExtra<DownloadedFile>(DOWNLOAD_BUNDLE)
+
+        if (downloadedFile.isSuccessful()) {
+            handleSuccess(downloadedFile)
+
+        } else if (downloadedFile.isFailed()) {
+
+            when (downloadedFile.reason) {
+                401, 403 -> handleUnauth(downloadedFile)
+                404 -> handleNotFound(downloadedFile)
+                else -> handleError(downloadedFile)
+            }
+
+        }
+
+        val downloadManager = getDownloadManager()
+        downloadManager.remove(downloadedFile.id)
+    }
+
+    private fun handleSuccess(downloadedFile: DownloadedFile) {
+
+        NotificationHelper.showProcessing(this, downloadedFile)
+
+        val downloadManager = getDownloadManager()
+        val uri = downloadManager.getUriForDownloadedFile(downloadedFile.id)
+
+        val inputStream = contentResolver.openInputStream(uri)
+
+        val outputFile = File(filesDir, downloadedFile.id.toString())
+        val outputStream = FileOutputStream(outputFile)
+
+        val buf = ByteArray(1024)
+        var len: Int = 0
+
+        while ({ len = inputStream.read(buf); len }() > 0) {
+            outputStream.write(buf, 0, len)
+        }
+
+        outputStream.close()
+        inputStream.close()
+
+        NotificationHelper.showFinished(this, downloadedFile)
+    }
+
+    private fun handleError(file: DownloadedFile) {
+        NotificationHelper.showError(this, file)
+    }
+
+    private fun handleNotFound(file: DownloadedFile) {
+        handleError(file)
```

Comment:
Shouldn't we at least display a specific message in these cases (not found and unauth)?